.github/workflows/ci.yml

```diff
@@ -191,88 +187,44 @@ jobs:
-        uses: google-github-actions/auth@v2
-        with:
-          credentials_json: ${{ secrets.GCP_SA_KEY }}
-
-      - name: Upload zcashd artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/zcashd{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
-
-      - name: Upload zcash-cli artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/zcash-cli{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
-
-      - name: Upload zcashd-wallet-tool artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/zcashd-wallet-tool{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
-
-      - name: Upload zcash-inspect artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/zcash-inspect{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
-
-      - name: Upload zcash-tx artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/zcash-tx{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
-
-      - name: Upload zcash-btest artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/test/test_bitcoin{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src/test
-
-      - name: Upload zcash-gtest artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/zcash-gtest{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
-
-      - name: Upload src/test/buildenv.py
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ./src/test/buildenv.py
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src/test
+      - name: Upload zcashd and related executables (src)
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ format('zcashd{0}', matrix.file_ext) }}-${{ matrix.name }}
+          path: |
+              ${{ format('./src/zcashd{0}', matrix.file_ext) }}
+              ${{ format('./src/zcash-cli{0}', matrix.file_ext) }}
+              ${{ format('./src/zcashd-wallet-tool{0}', matrix.file_ext) }}
+              ${{ format('./src/zcash-inspect{0}', matrix.file_ext) }}
+              ${{ format('./src/zcash-tx{0}', matrix.file_ext) }}
+              ${{ format('./src/test/test_bitcoin{0}', matrix.file_ext) }}
+              ${{ format('./src/zcash-gtest{0}', matrix.file_ext) }}
```

Comment:
In the original GitHub Actions artifacts approach (that was replaced by GCA in be26acd7456ee7edee7ae2e31f035f6c4beb1942), these particular binaries were stored in individual artifacts (with the rationale that we only needed one specific binary in the corresponding test jobs, so this would speed up the individual runners). Similarly, the other binaries were grouped based on what combinations were needed for testing. Does it work out to be simpler or faster to store all the binaries in one artifact?

Reply:
Yes, exactly, in the previous approach, we were optimizing. @daira had suggested combining them, which simplifies the code quite a bit. I'm satisfied with the time we've achieved, and I like the simplification. What I propose is to keep it for the sake of code simplicity/maintainability, and if in the future we see that the prices/time increase significantly or we're not satisfied, we can revisit it.